sdk/keyvault/azure-keyvault-administration/tests/test_backup_client.py

```diff
@@ -2,82 +2,100 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from functools import partial
 import time
+from functools import partial
 
+import pytest
 from azure.core.exceptions import ResourceExistsError
 from azure.keyvault.administration._internal import parse_folder_url
-import pytest
+from devtools_testutils import recorded_by_proxy, set_bodiless_matcher
 
 from _shared.test_case import KeyVaultTestCase
-from _test_case import AdministrationTestCase, backup_client_setup, get_decorator
-
+from _test_case import KeyVaultBackupClientPreparer, get_decorator
 
 all_api_versions = get_decorator()
 
 
-class BackupClientTests(AdministrationTestCase, KeyVaultTestCase):
-    def __init__(self, *args, **kwargs):
-        super(BackupClientTests, self).__init__(*args, match_body=False, **kwargs)
+class TestBackupClientTests(KeyVaultTestCase):
 
-    @all_api_versions()
-    @backup_client_setup
-    def test_full_backup_and_restore(self, client):
+    def create_key_client(self, vault_uri, **kwargs):
+        from azure.keyvault.keys import KeyClient
+        credential = self.get_credential(KeyClient)
+        return self.create_client_from_credential(KeyClient, credential=credential, vault_url=vault_uri, **kwargs )
+
+    @pytest.mark.parametrize("api_version", all_api_versions)
+    @KeyVaultBackupClientPreparer()
+    @recorded_by_proxy
+    def test_full_backup_and_restore(self, client, **kwargs):
+        set_bodiless_matcher()
         # backup the vault
-        backup_poller = client.begin_backup(self.container_uri, self.sas_token)
+        container_uri = kwargs.pop("container_uri")
+        sas_token = kwargs.pop("sas_token")
+        backup_poller = client.begin_backup(container_uri, sas_token)
         backup_operation = backup_poller.result()
         assert backup_operation.folder_url
 
         # restore the backup
-        restore_poller = client.begin_restore(backup_operation.folder_url, self.sas_token)
+        restore_poller = client.begin_restore(backup_operation.folder_url, sas_token)
         restore_poller.wait()
         if self.is_live:
             time.sleep(60)  # additional waiting to avoid conflicts with resources in other tests
 
-    @all_api_versions()
-    @backup_client_setup
-    def test_full_backup_and_restore_rehydration(self, client):
-        if not self.is_live:
-            pytest.skip("Poller requests are incompatible with vcrpy in playback")
+    @pytest.mark.parametrize("api_version", all_api_versions)
+    @KeyVaultBackupClientPreparer()
+    @recorded_by_proxy
+    def test_full_backup_and_restore_rehydration(self, client, **kwargs):
+        set_bodiless_matcher()
+        container_uri = kwargs.pop("container_uri")
+        sas_token = kwargs.pop("sas_token")
+
+        # if not self.is_live:
+        #     pytest.skip("Poller requests are incompatible with vcrpy in playback")
```

Comment:
It's great that these tests work in playback with the proxy! We can remove these comments in formerly-skipped tests now